Tests/Realm.Tests/Database/RealmResults/SimpleLINQtests.cs

```diff
@@ -686,16 +686,36 @@ public void StringSearch_Contains_CaseSensitivityTests()
             MakeThreePatricks();
 
             // case sensitive
-            var contains_atri = _realm.All<Person>().Where(p => p.FirstName.Contains("atri")).Count();
-            Assert.That(contains_atri, Is.EqualTo(2));
+            Assert.That(_realm.All<Person>().Where(p => p.FirstName.Contains("atri")).Count(), Is.EqualTo(2));
+            Assert.That(_realm.All<Person>().Where(p => p.FirstName.Contains("atri")).ToArray().Length, Is.EqualTo(2));
```

Comment:
Why is this kind of double check necessary?

Reply:
Not sure - I noticed the reported issue didn't repro when using `Count` but it reproduces when using anything that materializes the results.

Reply:
Mmm I see